ethexe/ethereum/src/mirror/events.rs

```diff
@@ -16,68 +16,65 @@
 // You should have received a copy of the GNU General Public License
 // along with this program. If not, see <https://www.gnu.org/licenses/>.
 
-use crate::IMirror;
-use alloy::{rpc::types::eth::Log, sol_types::SolEvent};
+use crate::{decode_log, IMirror};
+use alloy::{primitives::B256, rpc::types::eth::Log, sol_types::SolEvent};
 use anyhow::Result;
 use ethexe_common::mirror;
-use gprimitives::H256;
+use signatures::*;
 
 pub mod signatures {
-    use super::{IMirror, SolEvent, H256};
+    use super::*;
 
-    pub const EXECUTABLE_BALANCE_TOP_UP_REQUESTED: H256 =
-        H256(IMirror::ExecutableBalanceTopUpRequested::SIGNATURE_HASH.0);
-    pub const MESSAGE_QUEUEING_REQUESTED: H256 =
-        H256(IMirror::MessageQueueingRequested::SIGNATURE_HASH.0);
-    pub const MESSAGE: H256 = H256(IMirror::Message::SIGNATURE_HASH.0);
-    pub const REPLY_QUEUEING_REQUESTED: H256 =
-        H256(IMirror::ReplyQueueingRequested::SIGNATURE_HASH.0);
-    pub const REPLY: H256 = H256(IMirror::Reply::SIGNATURE_HASH.0);
-    pub const STATE_CHANGED: H256 = H256(IMirror::StateChanged::SIGNATURE_HASH.0);
-    pub const VALUE_CLAIMED: H256 = H256(IMirror::ValueClaimed::SIGNATURE_HASH.0);
-    pub const VALUE_CLAIMING_REQUESTED: H256 =
-        H256(IMirror::ValueClaimingRequested::SIGNATURE_HASH.0);
+    crate::signatures_consts! {
+        IMirror;
+        EXECUTABLE_BALANCE_TOP_UP_REQUESTED: ExecutableBalanceTopUpRequested,
+        MESSAGE_QUEUEING_REQUESTED: MessageQueueingRequested,
+        MESSAGE: Message,
+        REPLY_QUEUEING_REQUESTED: ReplyQueueingRequested,
+        REPLY: Reply,
+        STATE_CHANGED: StateChanged,
+        VALUE_CLAIMED: ValueClaimed,
+        VALUE_CLAIMING_REQUESTED: ValueClaimingRequested,
+    }
 
-    pub const ALL: [H256; 8] = [
+    pub const FOR_HANDLING: &[B256] = &[
         EXECUTABLE_BALANCE_TOP_UP_REQUESTED,
         MESSAGE_QUEUEING_REQUESTED,
-        MESSAGE,
         REPLY_QUEUEING_REQUESTED,
-        REPLY,
-        STATE_CHANGED,
-        VALUE_CLAIMED,
         VALUE_CLAIMING_REQUESTED,
```

Comment:
MInor: Why do we have naming like "_REQUESTED" in `sol` and naming "_FOR_HANDLING" in `rust`. Why not just use naming REQUESTED or REQUEST anywhere: `BlockEventRequest`, `RouterEventRequest` ?

Reply:
EventRequest sounds like request for event, but in fact its event that requests for handling (processing), wdyt?

Reply:
The main idea: we should use in rust the corresponding naming as in .sol